core/templates/dev/head/forms/formBuilder.js

```diff
@@ -672,7 +672,13 @@ oppia.factory('rteHelperService', [
           elt.find(
             'img.oppia-noninteractive-' + componentDefn.name
           ).replaceWith(function() {
-            var jQueryElt = $('<' + this.className.split(' ')[0] + '/>');
+            // Look for a class name starting with oppia-noninteractive-*.
+            var tagNameMatch = /(^|\s)(oppia-noninteractive-\w+)/.exec(
```

Comment:
One gotcha here: I think \w is incorrect, and should be replaced with something that means "lowercase chars and hyphens" -- otherwise this may break in the future for RTE components with multi-word names, and it will take a while to debug. Would something like `[a-z\-]+` work?